### PR TITLE
Updating run_cart.sh to work with sockets and verbs

### PR DESCRIPTION
**Pull request by sylviachanoiyee**

### CODE REVIEW

File: run_cart_sockets.sh [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
-
 #Parameter to be updated for each sbatch
 N_SERVERS=2
 N_CLIENTS=1
 M_SERVERS=8
 MAX_INFLIGHT=16
 CARTDIR="/home1/<PATH_TO_CART>/cart"
 
-interface=ib0
+interface=em1
```
Comment (saurabhtandan) [inline run_cart_sockets.sh:24]: Specific reason ib0 was removed? If I understand right, we can run sockets with ib0 and have a bit quicker run isn't it?

Comment (sylviachanoiyee) [reply run_cart_sockets.sh:24]: yep, good catch. will update in patch2.

File: run_cart_sockets.sh [hunk 2]
```diff
@@ -54,24 +54,24 @@ function wait {
 # run_server np cmd
 function run_bg {
         echo "run_bg CMD:"
-        cmd="orterun --mca btl tcp,self --mca oob tcp --mca mtl ^psm2,ofi --mca pml ob1 --map-by node --timeout $timeout --np $1 --hostfile $2 --output-filename testLogs/$4/sep_$test_$sep -x D_LOG_FILE=testLogs/$4/sep_$test_$sep/srv_output.log -x D_LOG_MASK=ERR -x CRT_PHY_ADDR_STR=\"ofi+verbs;ofi_rxm\" -x OFI_INTERFACE=$interface -x OFI_DOMAIN=mlx5_0  -x CRT_CTX_SHARE_ADDR=$sep -x CRT_CTX_NUM=16 -x PATH -x LD_LIBRARY_PATH $3 &"
+        cmd="orterun --mca btl self,tcp --map-by node --timeout $timeout --np $1 --hostfile $2 --output-filename testLogs/$4/sep_$test_$sep -x FI_LOG_LEVEL=debug  -x D_LOG_FILE=testLogs/$4/sep_$test_$sep/srv_output.log -x D_LOG_MASK=ERR -x CRT_PHY_ADDR_STR=\"ofi+sockets\" -x OFI_INTERFACE=$interface -x CRT_CTX_SHARE_ADDR=$sep -x CRT_CTX_NUM=16 -x PATH -x LD_LIBRARY_PATH $3 &"
```
Comment (ravalsam) [inline run_cart_sockets.sh:57]: Do we really need FI_LOG_LEVEL=debug in this script?

File: run_cart_sockets.sh [hunk 3]
```diff
@@ -54,24 +54,24 @@ function wait {
...
         eval $cmd
-        wait 5
+        wait 60
 }
 
 # run_server np cmd
 function run_fg {
         echo "run_fg CMD:"
-        cmd="orterun --mca btl tcp,self --mca oob tcp --mca mtl ^psm2,ofi --mca pml ob1 --map-by node --timeout $timeout --np $1 --hostfile $2 --output-filename testLogs/$4/sep_$test_$sep -x D_LOG_FILE=testLogs/$4/sep_$test_$sep/srv_output.log -x D_LOG_MASK=ERR -x CRT_PHY_ADDR_STR=\"ofi+verbs;ofi_rxm\" -x OFI_INTERFACE=$interface -x OFI_DOMAIN=mlx5_0  -x CRT_CTX_SHARE_ADDR=$sep -x CRT_CTX_NUM=16 -x PATH -x LD_LIBRARY_PATH $3"
+        cmd="orterun --mca btl self,tcp --map-by node --timeout $timeout --np $1 --hostfile $2 --output-filename testLogs/$4/sep_$test_$sep -x FI_LOG_LEVEL=debug -x D_LOG_FILE=testLogs/$4/sep_$test_$sep/srv_output.log -x D_LOG_MASK=ERR -x CRT_PHY_ADDR_STR=\"ofi+sockets\" -x OFI_INTERFACE=$interface -x CRT_CTX_SHARE_ADDR=$sep -x CRT_CTX_NUM=16 -x PATH -x LD_LIBRARY_PATH $3"
```
Comment (ravalsam) [inline run_cart_sockets.sh:68]: Comment same as above.

File: run_cart_sockets.sh (hunk 2 above)
Comment (sylviachanoiyee) [reply run_cart_sockets.sh:57]: I think it makes sense for this cart script. Very often than not, we run cart mostly for debugging purpose. But I can make it optional.

**Comment by sylviachanoiyee** [comment]
> I think this scripts can combine. Interface can be set on Top and other variables can be selected based on interface define by user. Except this two changes I do not see any major difference and same script can work for both providers, instead of maintain it both.

Yep, planning to combine it. Wanted to provide the tested scripts for cart folks to checkout asap.

**Pull request merged**